Earthfile

```diff
@@ -407,63 +402,34 @@ base-image:
-    # https://github.com/kairos-io/kairos-agent/blob/0288fb111bc568a1bfca59cb09f39302220475b6/pkg/elemental/elemental.go#L548   q
-    IF [ "$FLAVOR" = "fedora" ] || [ "$FLAVOR" = "rockylinux" ] || [ "$FLAVOR" = "almalinux" ]
-        RUN rm -rf /boot/initramfs-*
-    END
-
-    IF [ ! -e "/boot/vmlinuz" ]
-        IF [ -e "/boot/vmlinuz-lts" ]
-            # Alpine provides the kernel under this name
-            RUN ln -sf /boot/vmlinuz-lts /boot/vmlinuz
-        END
-        IF [ -e "/boot/vmlinuz-rpi4" ]
-            # Alpine-rpi provides the kernel under this name
-            RUN ln -sf /boot/vmlinuz-rpi4 /boot/vmlinuz
-        END
-        # If it's an ARM flavor, we want a symlink here from zImage/Image
-        # Check that its not a symlink already or grub will fail!
-        IF [ -e "/boot/Image" ] && [ ! -L "/boot/Image" ]
-            RUN ln -sf Image /boot/vmlinuz
-        ELSE IF [ -e "/boot/zImage" ]
-            IF  [ ! -L "/boot/zImage" ]
-                RUN ln -sf zImage /boot/vmlinuz
-            ELSE
-                RUN kernel=$(ls /boot/zImage-* | head -n1) && if [ -e "$kernel" ]; then ln -sf "${kernel#/boot/}" /boot/vmlinuz; fi
-            END
-        ELSE
-            # Debian has vmlinuz-VERSION
-            RUN kernel=$(ls /boot/vmlinuz-* | head -n1) && if [ -e "$kernel" ]; then ln -sf "${kernel#/boot/}" /boot/vmlinuz; fi
-            RUN kernel=$(ls /boot/Image-* | head -n1) && if [ -e "$kernel" ]; then ln -sf "${kernel#/boot/}" /boot/vmlinuz; fi
-        END
-    END
+    # Delete not needed initramfs files
+    RUN rm -rf /boot/initrd.img-* || true
+    RUN rm -rf /boot/initramfs-* || true
+    RUN --no-cache kernel=$(ls /lib/modules | head -n1) && depmod -a "${kernel}"
+    # Recreate initrd and link it to /boot/initrd
+    RUN --no-cache if [ -f "/usr/bin/dracut" ]; then kernel=$(ls /lib/modules | head -n1) && dracut -f "/boot/initrd-${kernel}" "${kernel}" && ln -sf "initrd-${kernel}" /boot/initrd;fi
+    RUN --no-cache if [ -f "/sbin/mkinitfs" ]; then kernel=$(ls /lib/modules | head -n1) && mkinitfs -o /boot/initrd $kernel; fi
+
+    # Create a symlink to the kernel to /boot/vmlinuz
+    RUN --no-cache if [ -e "/boot/vmlinuz-lts" ]; then ln -sf /boot/vmlinuz-lts /boot/vmlinuz; fi
+    RUN --no-cache if [ -e "/boot/vmlinuz-rpi4" ]; then ln -sf /boot/vmlinuz-rpi4 /boot/vmlinuz; fi
+    RUN --no-cache if [ -e " /boot/Image" ]; then ln -sf  /boot/Image /boot/vmlinuz; fi
+    RUN --no-cache if [ -e "/boot/zImage" ]; then ln -sf /boot/zImage /boot/vmlinuz; fi
+    RUN --no-cache kernel=$(ls /boot/zImage-* 2> /dev/null | head -n1) && if [ -e "$kernel" ]; then ln -sf "${kernel#/boot/}" /boot/vmlinuz; fi
+    RUN --no-cache kernel=$(ls /boot/vmlinuz-* 2> /dev/null | head -n1) && if [ -e "$kernel" ]; then ln -sf "${kernel#/boot/}" /boot/vmlinuz; fi
+    RUN --no-cache kernel=$(ls /boot/Image-* 2> /dev/null | head -n1) && if [ -e "$kernel" ]; then ln -sf "${kernel#/boot/}" /boot/vmlinuz; fi
```

Comment:
Eventually we will have to move these to the dockerfiles and it would be good to remember which line is there for which flavor. Maybe keep that information around as comments? We can always look back in this commit but I'm afraid more changes in the future might make it hard to discover.

It should be too long until we move them though, so maybe it's fine as it is. I'm not sure.

Reply:
Is it? Or should we do ALL the lines and then fail if there is no link created? 

I mean, I worked them so they wont run unless it finds the file so they are mostly non-op and only one triggers.

This means we can add all of them to all the dockerfiles and they will work everywhere. Would be nicer if it was some script or whatever, but as pure docker goes, this should be valid for all the flavors in existance and should not fail anywhere.

Reply:
we can wrap these to a script you are right. But were does this script belong? Maybe it should be in the framework image. If we don't want it in the final images, we can just delete it in the dockerfiles after running it.